lib/utils/whitespaceChecker.js

```diff
@@ -215,20 +216,20 @@ module.exports = function whitespaceChecker(targetWhitespace, expectation, messa
 		const oneCharBefore = source[index - 1];
 		const twoCharsBefore = source[index - 2];
 
-		if (!isValue(oneCharBefore)) {
+		if (isNullish(oneCharBefore)) {
 			return;
 		}
 
 		if (
 			targetWhitespace === 'space' &&
 			oneCharBefore === ' ' &&
-			(activeArgs.onlyOneChar || !isWhitespace(twoCharsBefore))
+			(activeArgs.onlyOneChar || isNullish(twoCharsBefore) || !isWhitespace(twoCharsBefore))
 		) {
 			return;
 		}
 
 		assertFunction(messageFunc);
-		activeArgs.err(messageFunc(activeArgs.errTarget || source[index]));
+		activeArgs.err(messageFunc(activeArgs.errTarget || source.charAt(index)));
```

Comment:
[note] `source[index]` can return `undefined`, while `source.charAt(index)` always returns a string. See the `charAt()` doc:

> If `index` is out of range, `charAt()` returns an empty string. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charAt

I think it safe to use `charAt()` in this case to build an error message.